src/sidebar/services/api.js

```diff
@@ -81,7 +99,7 @@ function get(object, path) {
  * @param {Promise<RouteMap>} links - API route data from API index endpoint (`/api/`)
  * @param {string} route - The dotted path of the named API route (eg. `annotation.create`)
  * @param {APIMethodCallbacks} callbacks
- * @return {APICall<Record<string, any>, object, object>} - Function that makes
+ * @return {APICall<Record<string, any>, Record<string, any>|void, unknown>} - Function that makes
  *   an API call. The returned `APICall` has generic parameter, body and return types.
  *   This can be cast to an `APICall` with more specific types.
```

Comment:
I'ma need a little help here. I'm having trouble unwinding how `APICall` works here. The comment here mentions "generic parameter, body and return types". Is that why we see `Record<string, any>` instead of `Params` and `Record<string, any>|void` for `Body`? Or...am I way off? And the `unknown`... does that map to a callback? I can't quite cobble these pieces together in my little beady brain, or understand why they need to be "generic" here.

Reply:
>  The comment here mentions "generic parameter, body and return types". Is that why we see Record<string, any> instead of Params and Record<string, any>|void for Body? 

Yes. In other words, this function returns an API call with very loose types (params is an object with unspecified properties, the body is either empty or an object, the result could be anything). This is then cast to an `APICall<...>` with more specific params, body and result types for each call.